api/v1alpha1/amaltheasession_types.go

```diff
@@ -99,15 +97,17 @@ type Session struct {
 type Ingress struct {
 	Annotations      map[string]string `json:"annotations,omitempty"`
 	IngressClassName string            `json:"ingressClassName,omitempty"`
-	Host             string            `json:"host,omitempty"`
-	PathPrefix       string            `json:"pathPrefix,omitempty"`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Host is immutable"
+	Host string `json:"host"`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="PathPrefix is immutable"
+	PathPrefix string `json:"pathPrefix,omitempty"`
 	// The name of the TLS secret, same as what is specified in a regular Kubernetes Ingress.
 	TLSSecretName string `json:"tlsSecretName,omitempty"`
 }
 
 type Storage struct {
 	ClassName string            `json:"storageClassName,omitempty"`
-	Size      resource.Quantity `json:"storageSize,omitempty"`
+	Size      resource.Quantity `json:"storageSize"`
```

Comment:
Hey so in my other PR #605 I made size default to `1Gi`. Is that OK? If someone is just trying to get things going it is nicer to require fewer defaults in the CR spec.

Reply:
Sure thing !

There might be some things that we need to think about in the light of [Optional VS Required](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required) from the API conventions.